zazu/dev/commands.py

```diff
@@ -230,19 +233,33 @@ def get_pulls_for_branch(branch):
 
 @dev.command()
 @click.pass_context
-def review(ctx):
+@click.option('--base', default='develop', help='The base branch to target')
+@click.option('--head', default='', help='The head branch (defaults to current branch)')
+def review(ctx, base, head):
     """Create or display pull request"""
-    import urllib
-    encoded_branch = urllib.quote_plus(ctx.obj.repo.active_branch.name)
+    # TODO(nwiles): Refactor this into a plugin.
     url = ctx.obj.repo.remotes.origin.url
-    start = 'github.com'
-    if start in url:
-        base_url = url[url.find(start):].replace('.git', '').replace(':', '/')
-        url = 'https://{}/compare/{}?expand=1'.format(base_url, encoded_branch)
-        click.echo('Opening "{}"'.format(url))
-        webbrowser.open_new(url)
-        # TODO: add link to ticket in the PR, zazu logo
-        # <img src="http://vignette1.wikia.nocookie.net/disney/images/c/ca/Zazu01cf.png" alt="Zazu" width=50"/>
+    if 'github.com' in url:
+        if not head:
+            head = ctx.obj.repo.active_branch.name
+        gh = zazu.github_helper.make_gh()
+        owner, repo_name = zazu.github_helper.parse_github_url(url)
+        repo = gh.get_user(owner).get_repo(repo_name)
+        existing_pulls = repo.get_pulls(state='open', base=base, head=head)
+        try:
+            pr = existing_pulls[0]
+        except IndexError:
+            click.echo('No existing review found, creating one...')
```

Comment:
Existing_pulls gives you all of the pulls, whats the reasoning behind just checking the first one in the list? 

For example. currently existing_pulls[0] returns : Increase code coverage to >60%, number 52 
That is not the review associated with this branch.

Consider the scenario: if i was to take over a branch (like this one) and did zazu dev review, it would not find the existing branch pull request, it would open the browser to the increasing code coverage >60% review (currently).

Even then, if you ran zazu dev review, you would also open up to the wrong pull.

If a pr exists -> check the body for fixes# -> match with head feature/## - correct pr found, else create
if no pr exists, create

let me know if i'm missing something.